turbopack/crates/turbo-tasks/src/message_queue.rs

```diff
@@ -15,15 +15,24 @@ const MAX_QUEUE_SIZE: usize = 256;
 type ArcMx<T> = Arc<Mutex<T>>;
 type CompilationEventChannel = mpsc::Sender<Arc<dyn CompilationEvent>>;
 
+#[derive(Debug, Clone, Eq, PartialEq, Hash, serde::Serialize, serde::Deserialize)]
+enum EventChannelType {
+    Global,
+    Type(String),
+}
+
 pub struct CompilationEventQueue {
     event_history: ArcMx<VecDeque<Arc<dyn CompilationEvent>>>,
```

Comment:
TBH I would remove the `event_history`. You need to subscribe before sending the events.